antsibull/docs_parsing/routing.py

```diff
@@ -0,0 +1,326 @@
+        for redirect_name, redirect_dst in symlink_redirects.items():
+            if redirect_name not in plugin_type_routing:
+                plugin_type_routing[redirect_name] = {}
+            if 'redirect' not in plugin_type_routing[redirect_name]:
+                plugin_type_routing[redirect_name]['redirect'] = redirect_dst
+            if plugin_type_routing[redirect_name]['redirect'] == redirect_dst:
+                plugin_type_routing[redirect_name]['redirect_is_symlink'] = True
+
+    if collection_name in COLLECTIONS_WITH_FLATMAPPING:
+        remove_flatmapping_artifacts(plugin_routing_out)
+
+    return plugin_routing_out
+
+
+async def load_all_collection_routing(collection_metadata: t.Mapping[str,
+                                                                     AnsibleCollectionMetadata]
+                                      ) -> t.Dict[str, t.Dict[str, t.Dict[str, t.Any]]]:
+    # Collection
+    lib_ctx = app_context.lib_ctx.get()
+    async with asyncio_pool.AioPool(size=lib_ctx.thread_max) as pool:
+        requestors = []
+        for collection, metadata in collection_metadata.items():
+            requestors.append(await pool.spawn(
+                load_collection_routing(collection, metadata)))
+
+        responses = await asyncio.gather(*requestors)
+
+    # Merge per-collection routing into one big routing table
+    global_plugin_routing: t.Dict[str, t.Dict[str, t.Dict[str, t.Any]]] = {}
+    for plugin_type in DOCUMENTABLE_PLUGINS:
+        global_plugin_routing[plugin_type] = {}
+        for collection_plugin_routing in responses:
+            global_plugin_routing[plugin_type].update(collection_plugin_routing[plugin_type])
+
+    return global_plugin_routing
+
+
+def remove_redirect_duplicates(plugin_info: t.MutableMapping[str, t.MutableMapping[str, t.Any]],
+                               collection_routing: CollectionRoutingT) -> None:
+    """
+    Remove duplicate plugin docs that come from symlinks (or once ansible-docs supports them,
+    other plugin routing redirects).
+    """
+    for plugin_type, plugin_map in plugin_info.items():
+        plugin_routing = collection_routing[plugin_type]
+        for plugin_name, plugin_record in list(plugin_map.items()):
+            # Check redirect
+            if plugin_name in plugin_routing and 'redirect' in plugin_routing[plugin_name]:
+                destination = plugin_routing[plugin_name]['redirect']
+                if destination in plugin_map and destination != plugin_name:
+                    # Heuristic: if we have a redirect, and docs for both this plugin and the
+                    # redirected one are generated from the same plugin filename, then we can
+                    # remove this plugin's docs and generate a redirect stub instead.
+                    if compare_all_but(plugin_record['doc'], plugin_map[destination]['doc'],
+                                       'filename'):
```

Comment:
```suggestion
                                       ['filename']):
```
Otherwise it ignores the keys `f`, `i`, `l`, ...